### ISSUE TITLE
MissingConstructor false positive with typed properties

### ISSUE THREAD

**Issue by simPod** [vimeo/psalm#3859]
I'm getting false positive MissingConstructor error. 

https://psalm.dev/docs/running_psalm/issues/MissingConstructor/

> Emitted when non-null properties without default values are defined in a class without a __construct method

I'm declaring nullable property `public ?string $id;` therefore I assume it should not trigger the error.

```php
<?php

declare(strict_types=1);

final class A
{
    public ?string $id;

    /** @param array{id?: string} $data */
    public static function fromArray(array $data) : self
    {
        $self = new self();
        $self->id = $data['id'] ?? null;

        return $self;
    }
}
```

https://psalm.dev/r/e3a1db5d6c

**Comment by simPod** [comment]
Ok, I have to set default value, right.
